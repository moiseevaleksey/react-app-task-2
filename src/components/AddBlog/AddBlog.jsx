import React from 'react';
import PropTypes from 'prop-types';

import convertFormDataToJson from '../utils/utils';
import http from '../../config/http';
import { API_BLOGS } from '../../config/api';

export default class AddBlog extends React.Component {
  postBlog = (blogFormData) => {
    blogFormData.append('date', new Date(Date.now()).toJSON());
    const blogData = convertFormDataToJson(blogFormData);
    http.post(API_BLOGS, blogData)
      .then(() => this.props.getBlogs())
      .catch(e => console.error(e));
  };

  handleSubmit = (e) => {
    e.preventDefault();
    const data = new FormData(this.blogForm);
    this.postBlog(data);
  };

  render() {
    return (
      <form
        onSubmit={e => this.handleSubmit(e)}
        ref={(blogForm) => { this.blogForm = blogForm; }}
      >
        <div className="input-group">
          <div className="input-group-prepend">
            <span className="input-group-text" id="">Add blog</span>
          </div>
          <input type="text" name="title" placeholder="Title" className="form-control" />
          <input type="text" name="author" placeholder="Author" className="form-control" />
          <input type="text" name="body" placeholder="Message" className="form-control" />
          <button className="btn btn-outline-secondary" type="submit">Add blog</button>
        </div>
      </form>
    );
  }
}

AddBlog.propTypes = {
  getBlogs: PropTypes.func.isRequired,
};
