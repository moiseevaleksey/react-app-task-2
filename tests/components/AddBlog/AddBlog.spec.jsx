import React from 'react';
import { render } from 'enzyme';
import toJson from 'enzyme-to-json';

import AddBlog from './../../../src/components/AddBlog/AddBlog';

describe('AddBlog', () => {
  let wrapper;
  const getBlogs = () => {};

  it('AddBlog should be rendered properly', () => {
    wrapper = render(<AddBlog getBlogs={getBlogs}/>);
    expect(toJson(wrapper)).toMatchSnapshot();
  });
});
