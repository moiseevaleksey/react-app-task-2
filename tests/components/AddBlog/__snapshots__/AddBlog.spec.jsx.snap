// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`AddBlog AddBlog should be rendered with props 1`] = `
<form>
  <div
    class="input-group"
  >
    <div
      class="input-group-prepend"
    >
      <span
        class="input-group-text"
        id=""
      >
        Add blog
      </span>
    </div>
    <input
      class="form-control"
      name="title"
      placeholder="Title"
      type="text"
    />
    <input
      class="form-control"
      name="author"
      placeholder="Author"
      type="text"
    />
    <input
      class="form-control"
      name="body"
      placeholder="Message"
      type="text"
    />
    <button
      class="btn btn-outline-secondary"
      type="submit"
    >
      Add blog
    </button>
  </div>
</form>
`;
